Sources/Logging/Logging.swift

```diff
@@ -13,7 +13,7 @@ public protocol LogHandler {
     subscript(metadataKey _: LoggingMetadata.Key) -> LoggingMetadata.Value? { get set }
 
     // All available metatdata
-    var metadata: LoggingMetadata? { get set }
+    var metadata: LoggingMetadata { get set } // removal of `?`, don't see the gain vs. using an empty map
```

Comment:
I see that `LoggingMetadata?` enabled performing a `set` with `nil` in order to work like a "clear" though I'm not entirely sure this is so nice, and we end up sprinkling a lot of `?` and `!` in logger handlers due to this.

How about we define a "clear context" as `= LoggingMetadata()` which is `= [:]` which should be fine and avoids all those `nil` compensating codepaths.